tests/PriorityMapTest.php

```diff
@@ -89,22 +91,33 @@ public function testGetKeys()
         $list->set('k4', 'v4', 1);
         $list->set('k5', 'v5', 4);
 
-        $this->assertEquals(
-            array('k1', 'k2', 'k3', 'k4', 'k5'),
-            $list->getKeys()
-        );
+        $result = $list->getKeys();
+
+        $this->assertCount(5, $result);
+        foreach($result as $key) {
+            $this->assertContains($key, $result);
+        }
+    }
+
+    public function testCount()
+    {
+        $list = new PriorityMap();
+
+        $list->set('K1', 'v1', 2);
+
+        $this->assertSame(1, $list->count());
     }
 
     public function testGetKeys_EmptyList()
     {
         $list = new PriorityMap();
 
-        $this->assertEquals(
-            array(),
-            $list->getKeys()
-        );
+        $this->assertCount(0, $list->getKeys());
```

Comment:
is this test really passed? `getKeys` for empty list returns 0?

Reply:
According to the ```getKeys``` method, it will return the result ```array_keys``` function.
And the ```array_keys``` will return the array type.
The empty array can let the ```assertCount``` return the ```0```.
So the test will not be failed.

Reply:
test is something that see method as blackbox. you don't know how it implemented. but you must check that `getKeys` really return array. if some refactoring will broke this method and it will start to return null on some cases, your test don't find this case.

Reply:
I have the two approaches to test the return array type really.
Please consider the following code:

```
 $this->assertCount(0, $list->getKeys());
 $this->assertInternalType('array');
```

```
 $this->assertSame([], $list->getKeys());
```

What do you think?